app/src/main/java/com/example/solanakotlincomposescaffold/usecase/AccountBalanceUseCase.kt

```diff
@@ -0,0 +1,45 @@
+package com.example.solanakotlincomposescaffold.usecase
+
+import android.net.Uri
+import android.util.Log
+import com.example.solanakotlincomposescaffold.networking.KtorHttpDriver
+import com.funkatronics.encoders.Base58
+import com.funkatronics.networking.Rpc20Driver
+import com.funkatronics.rpccore.JsonRpc20Request
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.add
+import kotlinx.serialization.json.buildJsonArray
+import java.util.UUID
+
+object AccountBalanceUseCase {
+    private val TAG = AccountBalanceUseCase::class.simpleName
+    suspend operator fun invoke(rpcUri: Uri, address: ByteArray): Long =
+        withContext(Dispatchers.IO) {
+            val rpc = Rpc20Driver(rpcUri.toString(), KtorHttpDriver())
```

Comment:
Is there any consideration for DI with`Rpc20Driver`? i.e. passing it as a parameter to each UseCase.

Reply:
yeah I thought about that, and might still change it later. But my reasoning was this: the Ktor Http driver being used here generates a new client for each request (so that we don't have to worry about closing the client manually). I don't have any concerns about the overhead of generating new client instances, tho this is certainly something we can improve here even if only for purely demonstration/best practices.